src/user/user.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@nestjs/common'
+import { InjectRepository } from '@nestjs/typeorm'
+import { User } from './user.entity'
+import { Repository } from 'typeorm'
+import { createUserDto } from './dto/create-user.dto'
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
```

Comment:
```suggestion
    private readonly userRepository: Repository<User>,
```
Marking it as readonly makes it clear that the repository should never be reassigned after dependency injection. It’s a best practice in NestJS and TypeScript to enforce immutability for injected dependencies and improve code safety and maintainability.